functions/Set-DbaCompression.ps1

```diff
@@ -0,0 +1,508 @@
+﻿function Set-DbaCompression {
+<#
+	.SYNOPSIS
+		Returns tables and indexes with preferred compression setting.
+
+     .DESCRIPTION
+		This function set the appropriate compression recommendation.
+        Remember Uptime is critical, the longer uptime, the more accurate the analysis is.
+        You would probably be best if you utilized Get-DbaUptime first, before running this command.
+		
+		Set-DbaCompression script derived from GitHub and the tigertoolbox 
+        (https://github.com/Microsoft/tigertoolbox/tree/master/Evaluate-Compression-Gains)
+	
+	.PARAMETER SqlInstance
+		SqlInstance name or SMO object representing the SQL Server to connect to. This can be a collection and recieve pipeline input
+	
+	.PARAMETER SqlCredential
+		PSCredential object to connect under. If not specified, current Windows login will be used.
+	
+	.PARAMETER Database
+		The database(s) to process - this list is autopopulated from the server. If unspecified, all databases will be processed.
+	
+	.PARAMETER ExcludeDatabase
+		The database(s) to exclude - this list is autopopulated from the server
+	
+	.PARAMETER IncludeSystemDBs
+		Switch parameter that when used will display system database information
+	
+	.PARAMETER Silent
+		Replaces user friendly yellow warnings with bloody red exceptions of doom!
+		Use this if you want the function to throw terminating errors you want to catch.
+    
+    .PARAMETER MaxRunTime
+		Will continue to Alter tables and indexes for the give amount of minutes.
+
+    .PARAMETER PercentCompression
+		Will only work on the tables/indexes that have the calulated savings at and higer for the given number provided.
+	
+	.NOTES
+		Author: Jason Squires (@js_0505, jstexasdba@gmail.com)
+		Tags: Compression, Table, Database
+		Website: https://dbatools.io
+		Copyright: (C) Chrissy LeMaire, clemaire@gmail.com
+		License: GNU GPL v3 https://opensource.org/licenses/GPL-3.0
+	
+	.LINK
+		https://dbatools.io/Set-DbaCompression
+	
+	.EXAMPLE
+		Set-DbaCompression -SqlInstance localhost -MaxRunTime 60 -PercentCompression 25
+		Set the compression run time to 60 minutes and will start the compression of of tables/indexes
+        that have a difference of 25% or higher between current and recommended.
+	
+	.EXAMPLE
+		Set-DbaCompression -SqlInstance ServerA -Database DBName -MaxRunTime 60 -PercentCompression 25 | Out-GridView
+		Set the compression run time to 60 minutes and will start the compression of of tables/indexes
+        that have a difference of 25% or higher between current and recommended and the results into and nicely formated GridView.
+	
+	.EXAMPLE
+		Set-DbaCompression -SqlInstance ServerA -MaxRunTime 60 -PercentCompression 25
+		Set the compression run time to 60 minutes and will start the compression of of tables/indexes; across all databases;
+        that have a difference of 25% or higher between current and recommended.
+	
```

Comment:
Need to add an example using the `-IncludeSystemDbs`, `-ExcludeDatabase` and `-SqlCredential` (just for completeness).